core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java

```diff
@@ -164,6 +162,13 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
         return new GeoDistanceFactory(aggregationName, vsParser.config(), InternalGeoDistance.FACTORY, origin, unit, distanceType, ranges, keyed);
     }
 
+    private static String getSearchParseExceptionMessage(String currentFieldName, String aggregationName) {
+        return "Failed when parsing field with name '" +  currentFieldName +
+                "' found inside aggregation with name '" + aggregationName + "'." + " Few Possible Causes for the error:" +
+                "1)Misspelling the field name.  2)Giving a field name which is not recognized by Geo Distance Aggregator." +
+                "3)Assigning the field to a value with unallowed object type. ";
```

Comment:
@jpountz  thanks for the review .... 

R u Sure u want that removed ? 
These causes evolved by seeing the code logic to [parse the input json](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java#L76-L164). This is an exhaustive list .its ok, I can remove it . But before that just for my satisfaction: Why do u think it is not helpful ? I thought that this will help user narrow down the problem. For eg the problem user faced in https://github.com/elastic/elasticsearch/issues/12391 . Even after looking at the term he might not get any clue that it might be a spelling error.
similar errors can arise for eg . 

```javascript
"ranges": [
                 {
                   "from": 100,
                   "to": 200
                 }
               ]

```
if he gives as

```javascript
"ranges":  {
                   "from": 100,
                   "to": 200
                 }
```
then he will get a similar error. If we go one step further and say y he might have faced this error, will it not help ? This is just like the "Did you mean help messages that is popular now a days "... giving a guess to guide the user.